math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -115,14 +115,35 @@ private void checkEdge(int e) {
         }
     }
 
+    private void ensureCapacity(int capacity) {
+        for (int i = vertices.size(); i <= capacity; ++i) {
+            vertices.add(null);
+            removedVertices.add(i);
+        }
+    }
+
+    @Override
+    public void addVertex(int v) {
+        if (v < 0) {
+            throw new PowsyblException("Invalid vertex index: " + v);
+        }
+
+        ensureCapacity(v + 1);
+        if (vertices.get(v) == null) {
+            vertices.set(v, new Vertex<>());
+            removedVertices.remove(v);
+        }
+    }
```

Comment:
Adjacency list invalidation and listener notification are missing here is case of vertex creation

Reply:
Indeed, I will fix it